core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -87,7 +87,8 @@ abstract class Chunk[+O] extends Serializable with ChunkPlatform[O] with ChunkRu
   def collect[O2](pf: PartialFunction[O, O2]): Chunk[O2] = {
     val b = makeArrayBuilder[Any]
     b.sizeHint(size)
-    foreach(o => if (pf.isDefinedAt(o)) b += pf(o))
+    val f = pf.runWith(b += _)
+    foreach { o => f(o); () }
```

Comment:
Well, this looks rather ugly. `pf.runWith` returns `O => Boolean` so we need to transform it to `O => Unit` to avoid warnings. This could be rewritten as follows (at the cost of extra lambda call):
```scala
foreach(pf.runWith(b += _).andThen(_ => ()))
```
@armanbilge, which one would you prefer?

Reply:
Ugly but performant is the right choice 😅